source/MaterialXGenShader/Nodes/CompoundNode.cpp

```diff
@@ -15,17 +15,22 @@ ShaderNodeImplPtr CompoundNode::create()
     return std::make_shared<CompoundNode>();
 }
 
-void CompoundNode::initialize(ElementPtr implementation, ShaderGenerator& shadergen)
+void CompoundNode::initialize(ElementPtr implementation, ShaderGenerator& shadergen, const GenOptions& options)
 {
-    ShaderNodeImpl::initialize(implementation, shadergen);
+    ShaderNodeImpl::initialize(implementation, shadergen, options);
 
     NodeGraphPtr graph = implementation->asA<NodeGraph>();
     if (!graph)
     {
         throw ExceptionShaderGenError("Element '" + implementation->getName() + "' is not a node graph implementation");
     }
 
-    _rootGraph = ShaderGraph::create(graph, shadergen);
+    // For compounds we do not want to publish all internal inputs
```

Comment:
I thought this would only be true if it was a node definition, otherwise if you just create a nodegraph then won't this hide all the unconnected inputs?

Reply:
Yes, but the CompundNode implementation here is only used for the case when a nodedef has a nodegraph implementation. And then the nodedef sets the interface and you don't want to expose any internals. 

The other case when a nodegraph is connected to a shaderref is not handled here, that is done during the initial traversal and construction of the shadergen graph.

Reply:
Okay. Wasn't sure. Makes sense to hide this then.